### ISSUE TITLE
feature: Add a way for users to dump/print pipeline env vars

### ISSUE THREAD

**Issue by edgarrmondragon** [meltano/meltano#7302]
### Feature scope

CLI (options, error messages, logging, etc.)

### Description

Previously, users could see all the environment variables that were passed to each plugin's process. These included env vars generated at runtime, such as `MELTANO_EXTRACTOR_NAME`.

This was removed by https://github.com/meltano/meltano/pull/7212.

It'd still be useful to have a (safe) way for users to check which variables are being passed, so I propose adding a `--dump-env` option to `meltano invoke` and `meltano run`:

```
meltano invoke --dump-env=/path/to/env.json tap-github
meltano run --dump-env=/path/to/env.json tap-github target-duckdb
```

**Comment by aaronsteers** [comment]
Related to:

- #7219 

@edgarrmondragon - Is there are strong reason to export as a json file vs user `--print-var` we have existing on `invoke`?

https://docs.meltano.com/reference/command-line-interface#debugging-plugin-environment

One challenge with outputting to a file is that in the `meltano run` scenarios, and to some lesser extent also `elt` scenarios, the file would need to contain an output for multiple plugins, since each can have its own env dictionary, and/or we'd want to dump one file per plugin called.

Wdyt?

**Comment by tayloramurphy** [comment]
I'd prefer this be closed and possibly considered as part of the spec in #7219

**Comment by edgarrmondragon** [comment]
@aaronsteers So, I didn't realize `--print-var` supported multiple instances, e.g. `--print-var=VAR1 --print-var=VAR2`.

> I'd prefer this be closed and possibly considered as part of the spec in #7219

Do you mean the ability to dump all env vars to a file?

**Comment by tayloramurphy** [comment]
@edgarrmondragon ah my mistake. I misread the scope of the proposal. Ignore my statement.

**Comment by aaronsteers** [comment]
> @aaronsteers So, I didn't realize --print-var supported multiple instances, e.g. --print-var=VAR1 --print-var=VAR2

Yeah, that was a nice implementation, I think. And flexible enough in how it is designed, that it would make sense and have sensible output even if multiple steps in the execution defined the variable differently.

Taking this example:

```
meltano run --print-var=AWS_ACCESS_KEY_ID --print-var=AWS_REGION tap-snowflake target-redshift dbt-redshift:run sqlfluff:lint
```

Presumably each plugin in the invocation could print its own values for `AWS_ACCESS_KEY_ID` and `AWS_REGION` - and the output would be meaningful and relevant.

```
Printing value for requested variable `AWS_ACCESS_KEY_ID` in plugin `tap-snowflake`: ASDF1234
Printing value for requested variable `AWS_REGION` in plugin `tap-snowflake`: us-west-2
// ...
Printing value for requested variable `AWS_ACCESS_KEY_ID` in plugin `target-redshift`: XYZA1234
Printing value for requested variable `AWS_REGION` in plugin `target-redshift`: us-east-1
// ...
Printing value for requested variable `AWS_ACCESS_KEY_ID` in plugin `dbt-redshift`: XYZA1234
Printing value for requested variable `AWS_REGION` in plugin `dbt-redshift`: us-east-1
```

The above output has possibility for many prints per env var, but collectively taken together, this likely could and would be helpful for debugging - for instance, to see that Snowflake and Redshift are in different regions, but the access key and region for target-redshift matches those of dbt-redshift.

Since `sqlfluff` doesn't use `AWS_ACCESS_KEY_ID` or `AWS_REGION`, this plugin could simply skip the print with no issue.
